File: civitasjava/src/Civitas/Sorpresa.java
package civitas;

import java.util.ArrayList;

public abstract class Sorpresa{
    private String texto;
    
    Sorpresa(String texto){
        this.texto = texto;
    }
    
    public boolean jugadorCorrecto(int actual, ArrayList<Jugador> todos){
        return actual <= todos.size();
    }
    
    
    protected void informe(int actual, ArrayList<Jugador> todos){
        String a = texto;
        String mensaje = "Se aplica una sorpresa " +a +" al jugador " 
                +todos.get(actual).getNombre();
        
        Diario.getInstance().ocurreEvento(mensaje);
    }
    
    public abstract void aplicarAJugador(int actual, ArrayList<Jugador> todos);
    
    @Override
    public String toString(){
        return "Texto: " +texto + "}";
    }
}

File: RubyCivitas/lib/sorpresa.rb
require_relative "tipo_sopresa"
require_relative "mazo_sorpresas"
require_relative "jugador"
require_relative "tablero"

module Civitas

  class Sorpresa

    def initialize (texto)
      @texto = texto
    end

    def jugador_correcto(actual, todos)
      
      return actual < todos.length && actual >= todos.find_index(todos.first)
      
    end

    def informe (actual, todos)

      Diario.instance.ocurre_evento("[DIARIO] Se esta aplicando una sorpresa a #{todos[actual]}")

    end

    def aplicar_al_jugador (actual, todos)
      
      informe(actual, todos)

    end

    def to_string

      return @texto

    end

  end
end
